src/rules_and_declarations.rs

```diff
@@ -483,10 +493,26 @@ where
 {
     let location = input.current_source_location();
     let delimiters = Delimiter::Semicolon | Delimiter::CurlyBracketBlock;
-    // FIXME: https://github.com/rust-lang/rust/issues/42508
-    let result = parse_until_before::<'i, 't, _, _, _>(input, delimiters, |input| {
-        parser.parse_prelude(name, input)
-    });
+
+    // Using the following closure to `parse_until_before` directly as an
+    // argument would require lifetime annotations to compile, to avoid a
+    // "cannot infer an appropriate lifetime due to conflicting requirements"
+    // error due to https://github.com/rust-lang/rust/issues/42508.
+    //
+    // However, doing so would then trigger a warning from the
+    // "late_bound_lifetime_arguments" lint, as the function has both
+    // "late-bound" and "early-bound" lifetime parameters.
+    //
+    // More information is available on the lint's tracking issue
+    // https://github.com/rust-lang/rust/issues/42868.
+    //
+    // Using this `callback` variable is a workaround for the lint's warning.
+    let result;
+    {
+        // Introduce a new scope to limit the duration of the `parser` loan
+        let callback = |input: &mut Parser<'i, '_>| parser.parse_prelude(name, input);
+        result = parse_until_before(input, delimiters, callback);
+    }
```

Comment:
Does the warning happen without assignment outside of `let`, like this?

```rust
let result = {
    …
};
```

If not, please change it and similarly `let prelude;` below.

Reply:
this also works indeed, done